test/unit/test_matrix_output.cpp

```diff
@@ -67,3 +76,174 @@ TYPED_TEST(MatrixOutputLocalTest, printElements) {
     printElements(mat);
   }
 }
+
+template <class T>
+struct is_complex : public std::false_type {};
+
+template <class T>
+struct is_complex<std::complex<T>> : public std::true_type {};
+
+template <class T>
+struct numpy_type;
+
+template <>
+struct numpy_type<float> {
+  static constexpr auto name = "float";
+};
+
+template <>
+struct numpy_type<double> {
+  static constexpr auto name = "float";
```

Comment:
I investigated a bit. In numpy, there are specific types for single and double precision with various synonyms. I quickly tried and there seems to be a problem with `np.complex128` (matching with std::complex<double>).

Anyway, I don't think that it is necessary to use the exact match for the type, and the single precision is way more than enough, since we are printing values with just few digits.

Do you agree @Sely85 @rasolca @teonnik ?

Reply:
Unless I am missing something, currently we are using the default precision which is 6. If we don't plan to increase the precision, then yes, it wouldn't make a difference if we have single and double precision.

Regardless, we should probably use `np.csingle` and `np.cdouble` instead of the fixed width aliases (`np.complex64` `np.complex128`) because `float` and `double` are technically platform-dependent.

Reply:
> Unless I am missing something, currently we are using the default precision which is 6. If we don't plan to increase the precision, then yes, it wouldn't make a difference if we have single and double precision.

Yes, it is what I meant in my previous comment.

> Regardless, we should probably use `np.csingle` and `np.cdouble` instead of the fixed width aliases (`np.complex64` `np.complex128`) because `float` and `double` are technically platform-dependent.
> 
> https://numpy.org/doc/stable/user/basics.types.html

I'm not completely getting your point. Let me know if you agree on this mapping (ctypes -> numpy):
- `float/double` -> **np.single** (_"Platform-defined single precision float"_)
- `complex<float/double>` -> **np.csingle** (_"Complex number, represented by two single-precision floats"_)

Reply:
Yep, I do.